DeleteUserDataActivity/backupAndDelete.ts

```diff
@@ -0,0 +1,472 @@
+import { BlobService } from "azure-storage";
+import { sequenceT } from "fp-ts/lib/Apply";
+import { Either, fromOption, left, toError } from "fp-ts/lib/Either";
+import { none, Option, some } from "fp-ts/lib/Option";
+import {
+  fromEither,
+  fromLeft,
+  TaskEither,
+  taskEither,
+  taskEitherSeq,
+  tryCatch
+} from "fp-ts/lib/TaskEither";
+
+import { array } from "fp-ts/lib/Array";
+import { MessageContent } from "io-functions-commons/dist/generated/definitions/MessageContent";
+import {
+  RetrievedMessageWithContent,
+  RetrievedMessageWithoutContent
+} from "io-functions-commons/dist/src/models/message";
+import { RetrievedMessageStatus } from "io-functions-commons/dist/src/models/message_status";
+import { RetrievedNotification } from "io-functions-commons/dist/src/models/notification";
+import { RetrievedNotificationStatus } from "io-functions-commons/dist/src/models/notification_status";
+import { RetrievedProfile } from "io-functions-commons/dist/src/models/profile";
+import { FiscalCode } from "italia-ts-commons/lib/strings";
+import {
+  IResultIterator,
+  iteratorToArray
+} from "../utils/extensions/documentdb";
+import { MessageDeletableModel } from "../utils/extensions/models/message";
+import { MessageStatusDeletableModel } from "../utils/extensions/models/message_status";
+import { NotificationDeletableModel } from "../utils/extensions/models/notification";
+import { NotificationStatusDeletableModel } from "../utils/extensions/models/notification_status";
+import { ProfileDeletableModel } from "../utils/extensions/models/profile";
+import { DataFailure, IBlobServiceInfo } from "./types";
+import { saveDataToBlob } from "./utils";
+import { toDocumentDeleteFailure, toQueryFailure } from "./utils";
+
+import {
+  fromQueryEither,
+  QueryError
+} from "io-functions-commons/dist/src/utils/documentdb";
+
+/**
+ * Recursively consumes an iterator and executes operations on every item
+ * @param deleteSingle takes an item and delete it
+ * @param userDataBackup references about where to save data
+ * @param makeBackupBlobName takes an item and construct a name for the backup blob
+ * @param iterator an iterator of every result from the db
+ */
+const executeRecursiveBackupAndDelete = <T>(
```

Comment:
It would be nice to generalize the recoursive logic and separate it from the specific use case (I.e the accumulators)